src/core/src/op/unsqueeze.cpp

```diff
@@ -82,29 +62,30 @@ bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out) {
     return true;
 }
 
-bool evaluate_unsqueeze(const HostTensorPtr& arg0, const HostTensorPtr& arg1, const HostTensorPtr& out) {
+bool evaluate_unsqueeze(const Node* node,
+                        const HostTensorPtr& arg0,
+                        const HostTensorPtr& arg1,
+                        const HostTensorPtr& out) {
     auto element_type = arg0->get_element_type();
     out->set_element_type(element_type);
 
-    auto data_shape = arg0->get_shape();
-    int64_t data_rank = static_cast<int64_t>(data_shape.size());
-    auto axes_shape = arg1->get_shape();
-    NGRAPH_CHECK(axes_shape.size() == 1 || axes_shape.empty(),
-                 "Axes to add must be a scalar or 1D tensor with 1 element");
+    const auto& axes_shape = arg1->get_shape();
+    OPENVINO_ASSERT(ov::is_scalar(axes_shape) || ov::is_vector(axes_shape),
+                    "Axes to add must be a scalar or 1D tensor with 1 element");
+
+    const auto& data_shape = arg0->get_shape();
+    const auto out_rank = static_cast<int64_t>(data_shape.size() + shape_size(axes_shape));
+
+    // Get axes and normalize
+    auto axes = read_index_vector(arg1);
+    normalize_axes(node, out_rank, axes);
```

Comment:
why can't we call shape_infer with axis passed as HostTensor?

Reply:
When shape inference use same as numpy checks for repeated axis then evaluate can use it added comment about it.